db/hybrid_logical_vector.go

```diff
@@ -399,3 +406,36 @@ func (hlv *HybridLogicalVector) setPreviousVersion(source string, version uint64
 	}
 	hlv.PreviousVersions[source] = version
 }
+
+// toHistoryForHLV formats blip History property for V4 replication and above
+func (hlv *HybridLogicalVector) toHistoryForHLV() string {
+	// take pv and mv from hlv if defined and add to history
+	var s strings.Builder
+	var separatorNeeded bool
+	// Merge versions must be defined first if they exist
+	if hlv.MergeVersions != nil {
+		var pairList []string
+		for key, value := range hlv.MergeVersions {
+			vrs := Version{SourceID: key, Value: value}
+			pairList = append(pairList, vrs.String())
+		}
+		if len(pairList) > 0 {
+			separatorNeeded = true
+			s.WriteString(strings.Join(pairList, ","))
```

Comment:
strings.Join is an efficient way to iterate over a set and do WriteString of separator and elements, but in this case we're already iterating over the set, and already have our own strings.Builder.  Why not just add elements to s as we iterate over MergeVersions and PreviousVersions, instead of iterating twice?

Reply:
I agree in hindsight this makes more sense. I originally went for this approach to avoid adding a trailing ',' at the end of the string as this would break the design CBL expect. I have got round that now with a item counter so we can keep track of the elements in the map we are iterating over.